x/btcstaking/keeper/msg_server.go

```diff
@@ -295,6 +294,13 @@ func (ms msgServer) CreateBTCDelegation(goCtx context.Context, req *types.MsgCre
 	if err := ms.AddBTCDelegation(ctx, newBTCDel); err != nil {
 		panic(fmt.Errorf("failed to add BTC delegation that has passed verification: %w", err))
 	}
+	// if this BTC delegation is restaked to consumers' FPs, add it to btcstkconsumer indexes
+	// TODO: revisit the relationship between BTC staking module and BTC staking consumer module
+	if restakedToConsumers {
+		if err := ms.IndexBTCConsumerDelegation(ctx, newBTCDel); err != nil {
```

Comment:
should it be indexed even if the BTC staking tx is not confirmed?

Reply:
Yes, this needs to be indexed at the beginning